File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {

        int dis = 0;
        int n = colors.length;

        for(int i = 0; i < n; i++){
            for(int j = i + 1; j < n; j++){
                if(colors[i] != colors[j]){
                    dis = Math.max(dis, j - i);
                }
            }
        }

        return dis;
    }
}

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.js
/**
 * @param {number[]} colors
 * @return {number}
 */
var maxDistance = function(colors) {
    let dis = 0;
    let n = colors.length;

    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            if (colors[i] !== colors[j]) {
                dis = Math.max(dis, j - i);
            }
        }
    }

    return dis;
};